packages/orm/src/EntityManager.ts

```diff
@@ -1884,7 +1898,7 @@ async function validateReactiveRules(
           e instanceof rule.source &&
           (e.isNewEntity ||
             e.isDeletedEntity ||
-            ((e as any).changes as Changes<any>).fields.some((f) => rule.fields.includes(f))),
+            ((e as any).changes as Changes<any>).fieldsWithoutRelations.some((f) => rule.fields.includes(f))),
```

Comment:
This ended up being another hot-spot b/c calculating "changed fields with m2o and m2m included" requires diffing the m2o/m2m, which is expensive, particularly the `entityCompare` sorter.